binder_compile.go

```diff
@@ -0,0 +1,164 @@
+package fiber
+
+import (
+	"bytes"
+	"encoding"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/gofiber/fiber/v3/internal/bind"
+	"github.com/gofiber/fiber/v3/utils"
+)
+
+type Decoder func(c Ctx, rv reflect.Value) error
+
+const bindTagRespHeader = "respHeader"
+const bindTagHeader = "header"
+const bindTagQuery = "query"
+const bindTagParam = "param"
+const bindTagCookie = "cookie"
+
+var textUnmarshalerType = reflect.TypeOf((*encoding.TextUnmarshaler)(nil)).Elem()
+var bindUnmarshalerType = reflect.TypeOf((*Binder)(nil)).Elem()
+
+func compileReqParser(rt reflect.Type) (Decoder, error) {
+	var decoders []decoder
+
+	el := rt.Elem()
+	if el.Kind() != reflect.Struct {
+		panic("wrapped request need to struct")
```

Comment:
I'll take a look here for map binding and multipart support.

Reply:
Oh, this panic need to be "return error", I forget this